plugins/github-actions/src/plugin.ts

```diff
@@ -23,10 +23,18 @@ import {
   githubAuthApiRef,
   createRoutableExtension,
   createComponentExtension,
+  createComponentAdaptationExtension,
 } from '@backstage/core-plugin-api';
+import { linkComponentRef } from '@backstage/core-components';
 
 export const githubActionsPlugin = createPlugin({
   id: 'github-actions',
+  adaptations: {
+    link: createComponentAdaptationExtension(linkComponentRef, {
+      id: 'github-actions-link-adaptation',
+      asyncAdaptation: () => import('./extensions/link').then(m => m.Provider),
```

Comment:
```suggestion
      asyncAdaptation: () => import('./adaptations/Link').then(m => m.Provider),
```

Wondering whether we should move the entire adaptation the separate file as the preferred pattern? Or do you think there's information here that's good to highlight?

Just for the purpose of keeping related information in the same place, no super strong opinion here. The dynamic import ofc gets a bit meh to define with an extra file but we can prolly handle that.